src/PollySandbox/Extensions/ResilienceExtensions.cs

```diff
@@ -0,0 +1,233 @@
+                StrategyName = builder.BuilderName,
+            });
+        }
+
+        static ValueTask<TimeSpan> ConstantWithJitter(RetryDelayArguments arguments, TimeSpan maxDelay)
+        {
+            double seed = arguments.DelayHint.TotalMilliseconds;
+            double max = maxDelay.TotalMilliseconds;
+
+            // From documentation here: https://github.com/App-vNext/Polly/wiki/Retry-with-jitter.
+            // Based on sample code given here: https://gist.github.com/reisenberger/a8d5806a95523f936d56a6d1dedc70a1.
+            // Adopting the "Decorrelated Jitter" formula from https://www.awsarchitectureblog.com/2015/03/backoff.html.
+            // Can be between seed and seed * 3. Must not exceed maxDelay.
+            double delay = Math.Min(max, Math.Max(seed, seed * 3 * Random.Shared.NextDouble()));
+
+            return ValueTask.FromResult(TimeSpan.FromMilliseconds(delay));
+        }
+    }
+
+    private static void ApplyTimeout(ResilienceStrategyBuilder builder, ApiEndpointOption endpoint)
+    {
+        builder.AddTimeout(new TimeoutStrategyOptions()
+        {
+            StrategyName = builder.BuilderName,
+            Timeout = endpoint.Timeout.Add(TimeSpan.FromSeconds(1)),
+        });
+    }
+
+    private static bool CanCircuitBreak(ApiException exception)
+    {
+        return
+            exception.StatusCode >= HttpStatusCode.InternalServerError ||
+            exception.StatusCode == HttpStatusCode.RequestTimeout;
+    }
+
+    private static bool CanRetry(ApiException exception)
+    {
+        if (exception.RequestMessage.Method != HttpMethod.Get)
+        {
+            return false;
+        }
+
+        return exception.StatusCode switch
+        {
+            HttpStatusCode.BadGateway => true,
+            HttpStatusCode.GatewayTimeout => true,
+            HttpStatusCode.RequestTimeout => true,
+            HttpStatusCode.ServiceUnavailable => true,
+            _ => false,
+        };
+    }
+
+    private static string[] GetEndpointNames<T>()
+        => typeof(T).GetMethods(BindingFlags.DeclaredOnly | BindingFlags.Instance | BindingFlags.Public)
+                    .Select((p) => p.Name)
+                    .ToArray();
+
+    private static void OnTelemetry(IMetricsPublisher publisher, TelemetryEventArguments args)
+    {
+        var bucket = new StringBuilder("polly.")
+            .Append(args.Source.StrategyType.ToLowerInvariant())
+            .Append('.')
+            .Append(args.Event.EventName.ToLowerInvariant());
+
+        if (args.Context.OperationKey is { Length: > 0 } key)
+        {
+            bucket.Append('.')
+                  .Append(key.ToLowerInvariant());
+        }
+
+        if (args.Arguments is ExecutionAttemptArguments attempt)
+        {
+            bucket.Append('.')
+                  .Append(attempt.Attempt);
+        }
+
+        publisher.Increment(bucket.ToString());
```

Comment:
`publisher.Increment(bucket.ToLowerInvariant())` and drop preceding `ToLowerInvariant` to not allocate intermediate strings.

Reply:
I did have that initially, but then changed it so that already lowered stuff wouldn't attempt to be relowered, but I didn't think about the extra allocations. Will change back.